src/test/java/com/fasterxml/jackson/databind/deser/std/CollectionDeserializer5139Test.java

```diff
@@ -0,0 +1,49 @@
+package com.fasterxml.jackson.databind.deser.std;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidNullException;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CollectionDeserializer5139Test {
+    static class Dst {
+        private List<Integer> list;
+
+        public List<Integer> getList() {
+            return list;
+        }
+
+        public void setList(List<Integer> list) {
+            this.list = list;
+        }
+    }
+
+    @Test
+    public void nullsFailTest() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configOverride(List.class).setSetterInfo(JsonSetter.Value.forContentNulls(Nulls.FAIL));
```

Comment:
Let's use `JsonMapper.builder()` approach so it merges more easily into 3.x (3.x has immutable `ObjectMapper`). 

Something like

```
mapper = JsonMapper.builder()
    .defaultSetterInfo(JsonSetter.Value.forContentNulls(Nulls.FAIL))
    .build();
```

Reply:
Fixed
93e1a21

Reply:
Oh, I am sorry.
There was a mistake and I have made an additional correction.
88bc7e4

Reply:
np, thanks!